Fix multiple label name usage in reused variables

jrgemignani opened the pull request:
Fixed an issue with clauses and chained clauses reusing variables with different label names attached. Now they are disallowed.

Fixed the function prototypes make_vertex_expr and make_edge_expr as they were not using the passed char *label parameter.

Added regression tests.

dehowef commented:
Plethora of regression tests, removal of previously unused label field. Looks good to me. I'll wait for @MuhammadTahaNaveed 's review before further action.

MuhammadTahaNaveed commented:
@dehowef @jrgemignani Just a small query here.`Labels that don't exist but do match` pass for edges but for vertices it fails. Should'nt that pass for vertices as well ?

```
MATCH (r1:invalid), (r1:invalid) return 0

ERROR:  multiple labels for variable 'r1' are not supported
LINE 2:         MATCH (r1:invalid), (r1:invalid) return 0
                                    ^
```

dehowef commented:
> @dehowef @jrgemignani Just a small question here.`Labels that don't exist but do match` pass for edges but for vertices it fails. Should'nt that pass for vertices as well ?
> 
> ```
> MATCH (r1:invalid), (r1:invalid) return 0
> 
> ERROR:  multiple labels for variable 'r1' are not supported
> LINE 2:         MATCH (r1:invalid), (r1:invalid) return 0
>                                     ^
> ```

it was probably overlooked, but yeah, I think the intended behavior is that it should fail for edges as well. Good looking out Taha. @jrgemignani, can you take a look when you have a chance?

MuhammadTahaNaveed commented:
@dehowef @jrgemignani Another issue I noticed later is the server termination caused by this query
```
MATCH (n:relateds),(n:related) return n
```
Here `relateds is an invalid label` and `related is a valid label`. This termination is most probably due to the missing condition `!cnode->label`.

jrgemignani commented:
> @dehowef @jrgemignani Another issue I noticed later is the server termination caused by this query
> 
> `MATCH (n:relateds),(n:related) return n`
> 

I'm not seeing a server termination -

```
psql-11.5-5432-pgsql=# SELECT * FROM cypher('graph1', $$ MATCH (n:relateds),(n:related) return n $$) as (r agtype);                 
ERROR:  multiple labels for variable 'n' are not supported
LINE 1: ...ECT * FROM cypher('graph1', $$ MATCH (n:relateds),(n:related...
                                                             ^
psql-11.5-5432-pgsql=#

```

MuhammadTahaNaveed commented:
> I'm not seeing a server termination -

@jrgemignani  I just verified and there is a termination at my end. I am attaching the prior commands as well.

```
SELECT create_graph('label_issue');
```
```
SELECT * FROM cypher('label_issue', $$
        CREATE (:related)                   
$$) AS (r0 agtype);
```
```
SELECT * FROM cypher('label_issue', $$
        MATCH (n:relateds),(n:related) return 0
$$) AS (r0 agtype);
```

jrgemignani commented:
@MuhammadTahaNaveed Does this crash exist outside of this PR?

jrgemignani commented:
@MuhammadTahaNaveed @dehowef I have updated my PR to cover vertices and edges for label reuse. I have also corrected the crash. Please verify, though.

MuhammadTahaNaveed commented:
> @MuhammadTahaNaveed @dehowef I have updated my PR to cover vertices and edges for label reuse. I have also corrected the crash. Please verify, though.

@jrgemignani Looks good. Great work.
